dali/base/dadfs.cpp

```diff
@@ -941,14 +943,48 @@ class FileClusterInfoArray: public IArrayOf<IClusterInfo>
                     for (i=0;i<clustlist.ordinality();i++)
                         if (strcmp(clustlist.item(i),cname.str())==0)
                             break;
-                    if (i==clustlist.ordinality())
+                    if (i==clustlist.ordinality()) // JCS - this will always be true if made it this far??
```

Comment:
no, because it could have broken out of the loop.  This tests if there was no match.